framework/src/play/cache/Cache.java

```diff
@@ -63,17 +73,38 @@ public static void clear() {
 
   /** Initialize the cache system. */
   public static void init() {
-    if ("enabled".equals(Play.configuration.getProperty("memcached", "disabled"))) {
+    Class<?> memcacheImplClass = null;
+    try {
+      memcacheImplClass = Class.forName("play.cache.MemcachedImpl");
+    } catch (ClassNotFoundException e) {
+      // Do nothing, leaving cacheImplClass null on purpose.
+    }
+
+    Class<?> ehCacheImplClass = null;
+    try {
+      ehCacheImplClass = Class.forName("play.cache.EhCacheImpl");
+    } catch (ClassNotFoundException e) {
+      // Do nothing, leaving cacheImplClass null on purpose.
+    }
+
+    if (memcacheImplClass != null && ehCacheImplClass != null) {
+      throw new ConfigurationException(
+          "Please remove either RePlay's `memcached` or `ehcache` package from the classpath");
+    }
+
+    Class<?> cacheImplClass = memcacheImplClass == null ? ehCacheImplClass : memcacheImplClass;
+    if (cacheImplClass != null) {
       try {
-        cacheImpl = new MemcachedImpl(Play.configuration);
-        logger.info("Connected to memcached");
-      } catch (Exception e) {
-        logger.error("Error while connecting to memcached", e);
-        logger.warn("Fallback to local cache");
-        cacheImpl = EhCacheImpl.newInstance();
+        // Since it implements the `CacheImpl` interface, it should have a static `instance` method.
+        Method method = cacheImplClass.getDeclaredMethod("instance", Properties.class);
+        cacheImpl = (CacheImpl) method.invoke(null, Play.configuration);
+      } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+        logger.error("Error while instantiating cache", e);
```

Comment:
I would throw an exception here instead of fall-back to dummy implementation. 
The sooner user gets aware of the problem the better.

Reply:
Changed this to throw an exception instead.